COMET/misc_plugins/PlotScripts/CONFIGS/importer/importer_PQC.py

```diff
@@ -0,0 +1,132 @@
+    capacity_dict = {}
+    area = {}
+    return_dict = {}
+
+
+    # Read in file content
+    for file in filepaths:
+        with open(file, "r") as fp:
+            file_content.append(fp.read())
+        file_name = filepaths[number_of_files].split('/')[-1]  #take the last element of the file path as the name of the file
+        dic[file_name] = file_content[number_of_files].split('\n') #assign the file name to its content in a dcitionary
+        dic[file_name]= [i.split('â')[0] for i in dic[file_name]] #sometimes for certain data there is an additional â that here is removed
+        dic[file_name] = list(filter(None, dic[file_name])) #remove blank lines from the data
+
+        index_of_datastart = 0
+        index_of_dataend = 0
+
+        #define two lists that will be used later to identify the line in the file where the date (12/10/2019) value is, and the start index of the data numerical values.
+        numbers = ['0','1','2','3','4','5','6','7','8','9','-']
+        data = ['2017','2018','2019','2020','2021','2022','2023','2024']
+
+        #find the index of the line in the file where the data values start
+        for element in dic[file_name]:
+
+            if element.split(' ')[0][-4:] not in data and len(element) > 1:  #element.split(' ')[0][-4:] checks that this is not a date entry(10/11/2019); len(element) > 1 checks that this is at least a two columns data value.
+                if element[0] not in numbers: #check if it is not a numerical value (non data)
+                    index_of_datastart += 1
+                    continue
+                else:
+                    index_of_datastart += 1
+                    start_data = index_of_datastart
+                    break
+
+        # find the index of the line in the file where the data values end
+        #reverse the list and repeat the above algorythm to find the start index of the data, in the end subtract this new find index to the index length of the file.
+        revlist = reversed(dic[file_name])
+
+        for element in revlist:
+
+            if element.split(' ')[0][-4:] not in data and len(element) > 1:
+                if element[0] not in numbers:
+                    index_of_dataend += 1
+                    continue
+                else:
+                    index_of_dataend += 1
+                    end_data = len(dic[file_name]) - index_of_dataend
+                    break
+
+
+        #importing from a gate diode file
+        if file_name[0:6] == 'IV_GCD':
+            volt = []
+            curr = []
+            for i in dic[file_name][start_data:(end_data + 1)]:
+                first, second = i.split('\t')   #extract elements of data from two columns and store the first element in a voltage list volt[] and the second in the current list curr[].
+                volt.append(first)
+                curr.append(second)
+            volt_dict[file_name] = [float(i) for i in volt] #assuring that the data are stored as float numbers
```

Comment:
What happens if there is a "a" inside? Then you get an error and your script crashes. A try except handler would be good here.